sympy/functions/elementary/trigonometric.py

```diff
@@ -1567,14 +1567,14 @@ def _eval_expand_trig(self, **hints):
             for i in range(n, -1, -1):
                 p[(n - i) % 2] += symmetric_poly(i, Y)*(-1)**(((n - i) % 4)//2)
             return (p[0]/p[1]).subs(list(zip(Y, CX)))
-        else:
+        elif arg.is_Mul:
             coeff, terms = arg.as_coeff_Mul(rational=True)
             if coeff.is_Integer and coeff > 1:
                 I = S.ImaginaryUnit
                 z = Symbol('dummy', real=True)
                 P = ((z + I)**coeff).expand()
                 return (re(P)/im(P)).subs([(z, cot(terms))])
-        return cot(arg)
```

Comment:
this is re-writing in terms of `tan`, not expanding, so this change shouldn't happen here. That will keep one of the tests from failing. The others just have to be edited. I suggest using `unchanged` to do the test and then add a test in which tests the Mul arg:
```python
>>> one = Symbol('one')
>>> def mul1(e):
...   n = e.atoms(Integer).pop()
...   return e.xreplace({n: n*one}).expand(trig=True).cancel()
...
>>> assert mul1(tan(4)) == (-4*tan(one)**3 + 4*tan(one))/(tan(one)**4 - 6*tan(one)**2 + 1)
True
```

Reply:
You could also use `Mul(4, 1, evaluate=False)` instead of `4`.